### PR TITLE
Rework 1-Wire implementation

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Rename OneWireControler to OneWireMaster.
- Add constructor to class to allow performing hardware & driver initializations in native.
- Class is now disposable to allow releasing resources in native.
- Add explanation about 1-Wire, configuration and usage samples to README.

## Motivation and Context
- Provide better usability.
- Allow better handling of init/uninit in native.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 -----
 
-### Welcome to the .NET **nanoFramework** 1-Wire Class Library repository
+### Welcome to the .NET **nanoFramework** 1-Wire&reg; Class Library repository
```
Comment (Ellerbach) [inline README.md]:
```suggestion
# Welcome to the .NET **nanoFramework** 1-Wire&reg; Class Library repository
```

File: nanoFramework.Device.OneWire/OneWireMaster.cs [hunk 2]
```diff
@@ -10,17 +10,52 @@
 namespace nanoFramework.Device.OneWire
 {
     /// <summary>
-    /// Represents a 1-Wire bus controller. The class provides methods and properties that an application can use to interact with the 1-Wire bus and connected devices.
+    /// Initializes a new instance of the <see cref="OneWireMaster"/> class.
+    /// Represents a 1-Wire bus master. The class provides methods and properties that an application can use to interact with the 1-Wire bus and connected devices.
     /// </summary>
-    public class OneWireController
+    public class OneWireMaster : IDisposable
```
Comment (Ellerbach) [inline nanoFramework.Device.OneWire/OneWireMaster.cs]: I would avoid the term 'Master' replace with Main or Principal or anything like this or even Controller.  Maybe even Host is the good word as it's the one used in the official documentation

File: nanoFramework.Device.OneWire/OneWireMaster.cs [hunk 3]
```diff
@@ -10,17 +10,52 @@
...
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OneWireMaster"/> class.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">If there is already another instance of the <see cref="OneWireMaster"/> class.</exception>
+        public OneWireMaster()
+        {
+            if (!s_opened)
+            {
+                _disposedValue = false;
```
Comment (Ellerbach) [inline nanoFramework.Device.OneWire/OneWireMaster.cs]: Maybe just disposed? I feel adding value is confusing

File: nanoFramework.Device.OneWire/OneWireMaster.cs (hunk 2 above)
Comment (josesimoes) [reply nanoFramework.Device.OneWire/OneWireMaster.cs]: Well... that's the name that's used in 1-Wire.... I get the thing about the wording but....

**Review by Ellerbach — COMMENTED** [review]
While making some braking changes here, why not making the API a bit more friendly and removing the 'Touch' from the names? Just 'Send' and 'Reset' will make a great job here!

File: nanoFramework.Device.OneWire/OneWireMaster.cs (hunk 3 above)
Comment (josesimoes) [reply nanoFramework.Device.OneWire/OneWireMaster.cs]: It's what we get on VS template for Dispose implementation 😅  but will change, no sweat.

**Comment by josesimoes** [comment]
> While making some braking changes here, why not making the API a bit more friendly and removing the 'Touch' from the names? Just 'Send' and 'Reset' will make a great job here!

Again, that's the names that are used in the 1-Wire interface and API... we can change those, but this won't be familiar to developers using the usual 1-Wire terminology...

**Review by Ellerbach — APPROVED** [review]
Thanks for the change! Looks good!

**Pull request merged**